### PR TITLE
[#4031] build: Add the asf yaml for Gravitino

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add the asf yaml for Gravitino

### Why are the changes needed?

Fix: #4031 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
This pull request needs to be verified after mering.

### CODE REVIEW

File: .asf.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+github:
+  description: World's most powerful open data catalog for building a high-performance, geo-distributed and federated metadata lake.
+  homepage: https://datastrato.ai/docs/0.5.1/
```
Comment (jerqi) [inline .asf.yaml:20]: This will be replaced when we have the new website.

Comment (yuqi1129) [inline .asf.yaml:20]: Suffix 0.5.1 is redundant.

Comment (jerqi) [reply .asf.yaml]: Done.

File: .asf.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - federated-query
+    - lakehouse
+    - model-catalog
+    - metalake
+    - skycomputing
+    - ai-catalog
+    - opendatacatalog
+  features:
+    # Enable wiki for documentation
+    wiki: true
```
Comment (jerryshao) [inline .asf.yaml:35]: We don't need to enable wiki, we will use website for documentation.

File: .asf.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    # Enable issues management
+    issues: true
+    # Enable projects for project management boards
+    projects: true
+  enabled_merge_buttons:
+    squash: true
+    merge: false
+    rebase: false
```
Comment (jerryshao) [inline .asf.yaml:43]: We'd better enable `rebase` for some scenarios like merging big features from dev branch.

File: .asf.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  protected_branches:
+    main:
+      required_status_checks:
+        strict: true
+      required_pull_request_reviews:
+        dismiss_stale_reviews: true
+        required_approving_review_count: 1
```
Comment (jerryshao) [inline .asf.yaml:50]: We should also add `branch-*` to the protected branches.

Comment (jerqi) [reply .asf.yaml:50]: I added.

File: .asf.yaml (hunk 3 above)
Comment (jerqi) [reply .asf.yaml]: I changed to true.

File: .asf.yaml (hunk 2 above)
Comment (jerqi) [reply .asf.yaml]: I changed it to false.

**Review by jerryshao — APPROVED** [review]
LGTM.

File: .asf.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+github:
+  description: World's most powerful open data catalog for building a high-performance, geo-distributed and federated metadata lake.
```
Comment (shaofengshi) [inline .asf.yaml:19]: The description may also need be updated after joing incubator. Using "Leading" or something else to replace "most powerful".

**Pull request merged**